ResearchKit/Common/ORKFormStep.m

```diff
@@ -81,6 +81,17 @@ - (void)validateParameters {
     for (ORKFormItem *item in _formItems) {
         [item.answerFormat validateParameters];
     }
+    
+    [self validateIdentifiersUnique];
+}
+
+- (void)validateIdentifiersUnique {
+    NSArray *uniqueIdentifiers = [_formItems valueForKeyPath:@"@distinctUnionOfObjects.identifier"];
+    BOOL itemsHaveNonUniqueIdentifiers = _formItems.count != uniqueIdentifiers.count;
+    
+    if (itemsHaveNonUniqueIdentifiers) {
+        @throw [NSException exceptionWithName:NSGenericException reason:@"All form items should have unique identifier" userInfo:nil];
```

Comment:
My (weak) preference is for the former, so that the underlying method call is clear. Certainly don't want to be making global changes to move to one or the other since they're completely equivalent.